src/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs

```diff
@@ -164,6 +165,7 @@ public partial struct AssemblyReference
         public System.Reflection.Metadata.StringHandle Name { get { throw null; } }
         public System.Reflection.Metadata.BlobHandle PublicKeyOrToken { get { throw null; } }
         public System.Version Version { get { throw null; } }
+        public System.Reflection.AssemblyName AssemblyName { get { throw null; } }
```

Comment:
Yes. The rationale here is that being consumed by very performance sensitive callers, this library is very careful to designate things which take time and allocate as methods, not properties.